src/main/java/com/google/sps/servlets/NotificationServlet.java

```diff
@@ -0,0 +1,194 @@
+    // Get user's ID from Utility method.
+    int userId = Utility.getUserId();
+    // Fetch notifications if user is signed in and convert the ArrayList to JSON
+    // using Utility method.
+    response.setContentType("application/json;");
+    response.getWriter().println(Utility.convertToJsonUsingGson(getNotifications(userId)));
+  }
+
+  /**
+   * Posts a new notification for all followers of a modified question or answer.
+   */
+  @Override
+  public void doPost(HttpServletRequest request, HttpServletResponse response) throws IOException {
+    // Define local time for the new entries in the server.
+    Timestamp localTimestamp = Timestamp.valueOf(LocalDateTime.now());
+    // Check if notification is about a question answered or answer commented, along with its id.
+    String typeOfNotification = request.getParameter("type");
+    int modifiedElementId = Integer.parseInt(request.getParameter("modifiedElementId"));
+
+    // URL to which the user will be redirected.
+    String notificationUrl = "";
+    String notificationMessage = "";
+    String query = "";
+    if (typeOfNotification.equals("question")) {
+      // If the notification is for an anwer to a question.
+      query =  "SELECT follower_id FROM QuestionFollower WHERE question_id = " + modifiedElementId;
+      notificationUrl = "/question.html?id=" + modifiedElementId;
+      notificationMessage = "Your question was answered.";
+    } else if (typeOfNotification.equals("answer")) {
+      // If the notification is for a new comment in an answer.
+      query =  "SELECT follower_id FROM AnswerFollower WHERE answer_id = " + modifiedElementId;
+      notificationUrl = "/question.html?id=" + modifiedElementId;
+      notificationMessage = "Your answer was commented.";
+    }
+    // Creates notification and relationship between its ID and the ID of the concerned users.
+    createNotification(query, notificationUrl, notificationMessage, localTimestamp);
+  }
+
+  /**
+   * Fetches notifications with the user ID.
+   */
+  private List<Notification> getNotifications(int userId) {
+    // Prepare query to select notifications by the subquery of notifications IDs selected by 
+    // user ID.
+    String query =  "SELECT message, url, date_time FROM Notification WHERE id IN " +
+                    "(SELECT notification_id FROM UserNotification WHERE user_id = " + userId +
+                    ") ORDER BY date_time DESC";
+    List<Notification> notifications = new ArrayList<>();
+    // Query the information from tables and create notification object to be stored in ArrayList.
+    try (Connection connection = DriverManager.getConnection(Utility.SQL_LOCAL_URL,
+                                                             Utility.SQL_LOCAL_USER,
+                                                             Utility.SQL_LOCAL_PASSWORD);
+         PreparedStatement pst = connection.prepareStatement(query);
+         ResultSet rs = pst.executeQuery()) {
```

Comment:
Prefer variable names that are more descriptive - for instance, if I see 'rs' later in the method, I have no idea what it is if I don't remember its initialization.

Reply:
done!